### PR TITLE
#654 - fix inconsistent sorting

### PR DESCRIPTION
**Pull request by Tiefseetauchner**
clamping down on sorting of songs so that the display is equivalent to the queue

### CODE REVIEW

**Comment by Tiefseetauchner** [comment]
Added Disabling of the top songs to the settings

File: lib/models/finamp_models.dart [hunk 1]
```diff
@@ -215,6 +217,9 @@ class FinampSettings {
   bool hideSongArtistsIfSameAsAlbumArtists =
       _hideSongArtistsIfSameAsAlbumArtists;
 
+  @HiveField(47, defaultValue: _showArtistsTopSongs)
+  bool showArtistsTopSongs = _showArtistsTopSongs;
+
```
Comment (Chaphasilor) [inline lib/models/finamp_models.dart:222]: Please move this to the end, so it's easier to figure out what the next ID should be

File: lib/components/ArtistScreen/artist_screen_content.dart [hunk 2]
```diff
@@ -112,11 +115,12 @@ class _ArtistScreenContentState extends State<ArtistScreenContent> {
...
-            sortedsongs.sort(
-              (a, b) =>
-                  b.userData?.playCount.compareTo(a.userData?.playCount ?? 0) ??
-                  0,
-            );
+            sortedsongs.sort((a, b) {
+              int compareCount = (b.userData?.playCount ?? 0)
+                  .compareTo(a.userData?.playCount ?? 0);
+              if (compareCount != 0) return compareCount;
+              return (a.name ?? "").compareTo(b.name ?? "");
```
Comment (Chaphasilor) [inline lib/components/ArtistScreen/artist_screen_content.dart:122]: There should be a `nameForSorting` attribute on the `BaseItemDto` to emulate Jellyfin's sorting behavior (e.g. excluding articles for albums, ignoring case, etc.), please compare that instead!

File: lib/l10n/app_de.arb [hunk 3]
```diff
@@ -301,6 +301,10 @@
   "@showCoverAsPlayerBackgroundSubtitle": {},
   "hideSongArtistsIfSameAsAlbumArtists": "Verstecke Titel-Künstler falls derselbe wie Album-Künstler",
   "@hideSongArtistsIfSameAsAlbumArtists": {},
+  "showArtistsTopSongs": "Meistgehörte Lieder bei Künstler anzeigen",
```
Comment (Chaphasilor) [inline lib/l10n/app_de.arb:304]: Maybe "Meistgehörte Lieder von Künstlern anzeigen" or "Meistgehörte Lieder auf Künstler-Seite anzeigen"?

File: lib/l10n/app_de.arb [hunk 4]
```diff
@@ -301,6 +301,10 @@
...
+  "@showArtistsTopSongs": {},
+  "showArtistsTopSongsSubtitle": "Ob oder nicht die meistgehörte Lieder bei Künstlern angezeigt werden sollen.",
```
Comment (Chaphasilor) [inline lib/l10n/app_de.arb:306]: "Ob die meistgehörten Lieder eines Künstlers angezeigt werden sollen, oder nicht."  
or maybe just
"Sollen die meistgehörten Lieder eines Künstlers (auf der Künstler-Seite) angezeigt werden?"

**Review by Chaphasilor — CHANGES_REQUESTED** [review]
Nice change overall, but I haven't looked at the actual app yet, just the code. I'll give it a try tomorrow.  
Just a few things that should be changed, but I could take care of those if you don't have the time...

**Comment by Chaphasilor** [comment]
While the correct song is selected, it seems like the content of the queue is not identical to the top songs list:
image
image

The queue should imo either be identical to the top song list, or should start a song mix, or, if you want to keep all artist songs part of the queue, the selected song should be at first position in the queue. What do you think?

**Comment by Tiefseetauchner** [comment]
I fully agree, I think that was a point of discussion, I'm fully for using the same list

File: lib/l10n/app_de.arb (hunk 4 above)
Comment (Tiefseetauchner) [reply lib/l10n/app_de.arb:306]: I would prefer the first one without the "oder nicht", I thought maybe that's just a Germany-exclusive colloquialism that I didn't know about, and I followed the other translations there. I'll change all occurrences as I hate the phrasing

**Comment by Chaphasilor** [comment]
I'm still getting inconsistent sorting if multiple tracks have not play count (e.g. all 0 plays). In that case, the selected song is always the last in queue, and the order of the other songs is inconsistent with the top songs.  

For artists where all songs have different play counts, it works well now 👍🏻

**Comment by Tiefseetauchner** [comment]
Yeah I was fighting with futures when I got distracted and tried to setup git extensions for the past hour, if you could have a look at how to solve that I'd greatly appreciate it, currently stuck on how to load all songs asynchronously

**Comment by Chaphasilor** [comment]
Alright. What I did was simply fetch *all* songs sorted by play count (didn't take much longer than just the first 5 on my phone, I think the page size of 100 is now used?), and only use the first (up to) 5 for the song songs section. We could now also have a "See All" button, or something similar.  

Let me know if this causes trouble on your phone with artists that have many songs! The only real bottleneck could be a slow connection or slow JSON parsing...

**Comment by Chaphasilor** [comment]
@Tiefseetauchner if everything works for you I can merge this :)

File: linux/flutter/generated_plugin_registrant.cc
Comment (Tiefseetauchner) [inline linux/flutter/generated_plugin_registrant.cc:1]: Does this belong in here?

File: linux/flutter/generated_plugin_registrant.h
Comment (Tiefseetauchner) [inline linux/flutter/generated_plugin_registrant.h:1]: Does this belong in here?

File: linux/flutter/generated_plugins.cmake
Comment (Tiefseetauchner) [inline linux/flutter/generated_plugins.cmake:1]: Does this belong in here?

File: macos/Flutter/ephemeral/Flutter-Generated.xcconfig
Comment (Tiefseetauchner) [inline macos/Flutter/ephemeral/Flutter-Generated.xcconfig:1]: Does this belong in here?

File: macos/Flutter/ephemeral/flutter_export_environment.sh
Comment (Tiefseetauchner) [inline macos/Flutter/ephemeral/flutter_export_environment.sh:1]: Does this belong in here?

**Review by Tiefseetauchner — COMMENTED** [review]
Seems like build tools from the desktop branch got commited

**Comment by Tiefseetauchner** [comment]
Sorry, I was distracted, seems to work well, even faster than before tho that might be placebo. You commited some desktop files, was that on purpose?

**Review by Chaphasilor — APPROVED** [review]

**Pull request merged**